src/util/generate-pnp-map-api.tpl.js

```diff
@@ -323,7 +323,7 @@ exports.resolveToUnqualified = function resolveToUnqualified(request, issuer, {c
   // contain multiple levels of dependencies (ie. a yarn.lock inside a subfolder of a yarn.lock). This is
   // typically solved using workspaces, but not all of them have been converted already.
 
-  if (ignorePattern && ignorePattern.test(issuer)) {
+  if (ignorePattern && ignorePattern.test(normalizePath(path.normalize(issuer)))) {
     const result = callNativeResolution(request, issuer);
```

Comment:
Depending on how the `pathRegExp` is used you might be able to gate on `path.parse()`